clientd3d/dialog.c

```diff
@@ -651,7 +651,18 @@ void DisplayDescription(object_node *obj, BYTE flags, char *description,
 	info.description  = description;
 	info.fixed_string = fixed_string;
 	info.url          = url;
+	info.rarity       = obj->rarity;
 	
+	std::string raritySuffix = GetRaritySuffix(info.rarity);
+
+	if (!raritySuffix.empty())
+	{
+        char nameWithSuffix[MAXNAME + 15];
+		snprintf(nameWithSuffix, sizeof(nameWithSuffix),
+		   "%s (%s)", info.name, raritySuffix.c_str());
+        info.name = nameWithSuffix;
```

Comment:
This is a serious bug.  info.name is pointing to nameWithSuffix, which is on the stack and immediately goes out of scope.  You will get garbage text and the game will sometimes crash when this memory is later overwritten.

One fix would be to make info,name a std::string, but I'm not sure how big a change that is.

Reply:
Converted info.name to std::string as suggested.